pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -329,9 +329,24 @@ func (l *LibvirtDomainManager) preStartHook(vmi *v1.VirtualMachineInstance, doma
 	// generate cloud-init data
 	cloudInitData := cloudinit.GetCloudInitNoCloudSource(vmi)
 	if cloudInitData != nil {
+		if cloudInitData.NetworkData == "" {
+			// get cloud-init network information
+			logger.Info("Starting cloud-init network discovery.")
+			cloudInitNetworkInfo, err := network.CloudInitDiscoverNetworkData(vmi)
+			if err != nil {
+				return domain, err
+			}
+
+			if len(cloudInitNetworkInfo) > 0 {
+				cloudInitData.NetworkData = string(cloudInitNetworkInfo)
```

Comment:
This will not overwrite any data been configure by the user?

Reply:
No `if cloudInitData.NetworkData == ""` means the user passed in data.